src/main/java/com/cloudbees/plugins/deployer/sources/MavenArtifactDeploySource.java

```diff
@@ -157,6 +158,7 @@ public String getGroupId() {
      * @return the type of artifact to match.
      */
     @NonNull
+    @SuppressFBWarnings(value = "RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE", justification = "Might be null if deserialized")
     public String getType() {
         return type == null ? DEFAULT_TYPE : type;
```

Comment:
Could remove the fb warning suppression with `StringUtils.defaultString(type, DEFAULT_TYPE)`